backend/migrating/v2/README.md

```diff
@@ -1,17 +1,113 @@
-# V2 Migrations and Commands
+# V2 Migration Guide
 
 This folder contains scripts and management commands for performing data migrations from v1 to v2 for the Unstract Multitenancy system.
+From version `v0.90.3` onwards, there have been signifcant changes in how data is stored. This warrants a data migration and the below steps can come in handy.
 
-## Overview
+## Preparing for the migration
 
-The commands provided here help you manage the migration process, including creating and dropping schemas and applying the necessary migrations to transition from v1 to v2.
+### Backup existing data
+
+- Make sure `pg_dump` is installed. This would usually be bundled with postgres and can be installed with
+
+```shell
+sudo apt-get install postgresql-client
+```
+
+- Take a backup of the existing data with the below command (provide the appropriate options)
+
+```shell
+pg_dump -h localhost -p 5432 -U unstract_dev -d unstract_db -f unstract-backup-$(date +"%Y-%m-%d_%H-%M-%S").sql
+```
+
+### Create a new database
+
+- Create a new database, for example `unstract_db_v2`
+
+### Dump existing data to this new database
+
+- Exec into the DB if required (from [/docker](/docker/))
+
+```shell
+docker exec -it unstract-db /bin/bash
+```
+
+- Execute the below command to restore the backup in the new DB
 
+```shell
+psql -h localhost -p 5432 -U unstract_dev -d unstract_db_v2 -f unstract-backup-<DateTime>.sql
+```
+
+### Using a feature flag
+
+- Check if this migration is used in combination with a feature flag. You can check the code for occurrences of any relevant [FeatureFlag](https://github.com/Zipstack/unstract/blob/0b9981514654634f8b198cdb416b1c9bcd6d423e/backend/backend/constants.py#L33) to confirm
+- If that is the case, head to Flipt configured at `http://localhost:8082` and add a new flag with the same name - in this example of `v2` it is `multi_tenancy_v2`
+
+## Performing the migration
+
+The script `migrate-to-v2.sh` in this directory can help perform the migration
+
+```shell
+./migrate-to-v2.sh
+```
+
+Follow the below steps to run each command necessary for the migration explicitly
+
+### Configuring the env
+
+- Make sure to update all the necessary `.env` files
+- For example, in case of `v2` ensure `backend`, `prompt-service`, and `platform-service` `.env` contain `DB_SCHEMA=unstract_v2`
```

Comment:
@chandrasekharan-zipstack Move this under `## Preparing for the migration` as the second last step.

Reply:
@hari-kuriakose I've restructured this `README.md` to pose the easier option first with relevant warnings.
Configuring env and creating a new schema are moved to a new section which lists the steps to perform the migration explicitly